src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs

```diff
@@ -114,7 +116,14 @@ public async Task<CompletionResponse> Handle(CompletionRequest request)
                 return new CompletionResponse { Items = ImmutableArray<CompletionItem>.Empty };
             }
 
-            var completions = await completionService.GetCompletionsAsync(document, position, getCompletionTrigger(includeTriggerCharacter: false));
+            OptionSet completionsWithImport = await document.GetOptionsAsync();
+            completionsWithImport = completionsWithImport.WithChangedOption(CompletionItemExtensions.ShowItemsFromUnimportedNamespaces, LanguageNames.CSharp, true);
```

Comment:
does this have to be changed per request? could we not set it as workspace option?

we already have infrastructure for binding `OmniSharpOptions` to workspace `OptionsSets` e.g. https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Roslyn.CSharp/Services/RenameWorkspaceOptionsProvider.cs This also takes care of watching for changes and updating the workspace if omnisharp.json changes. It seems to me like a way to go here

Reply:
I think I've correctly done this. Let me know if I messed something up.